src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java

```diff
@@ -3341,6 +3341,86 @@ public NBTTagCompound getOutput(Triple<NBTTagCompound, String, OperatorBase.Safe
                 return ValueObjectTypeRecipe.ValueRecipe.of(null);
             }).build());
 
+    /**
+     * ------------------------------------ PARSE OPERATORS ------------------------------------
+     */
+
+    /**
+     * Boolean Parse operator which takes a string of form `/(F(alse)?|[+-]?(0x|#)?0+|)/i`.
+     */
+    public static final IOperator PARSE_BOOLEAN = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.BOOLEAN, v -> {
+      ValueTypeString.ValueString value = v.getValue(0);
+      java.util.regex.Pattern p = java.util.regex.Pattern.compile("\\A(F(alse)?|[+-]?(0x|#)?0+|)\\z", java.util.regex.Pattern.CASE_INSENSITIVE);
+      if (p.matcher(value.getRawValue().trim()).matches()){
```

Comment:
This block can be simplified to `ValueTypeBoolean.ValueBoolean.of(p.matcher(value.getRawValue().trim()).matches())`

Reply:
I'd avoid changing the implementations of `deserialize`, to avoid unneeded overhead.
The regexes here are fine IMO.